### PR TITLE
minor update to ReleaseDuty docs

### PR DESCRIPTION
**Pull request by gbrownmozilla**
* remove obsolete links
 * clarify meetings and calendars
 * standardize on "ReleaseDuty" spelling

### CODE REVIEW

File: procedures/release-duty/index.rst [hunk 1]
```diff
@@ -65,25 +65,21 @@ Email
 
 As ReleaseDuty you need to *subscribe* to certain mailing lists:
 
- * All types of sign-offs and approvals should go to `release signoff mailing list <https://mail.mozilla.org/listinfo/release-signoff>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:68]: Can we revert this change. The mailinglists have been deprecated in favor of google groups - https://groups.google.com/a/mozilla.org/g/release-signoff/

File: procedures/release-duty/index.rst [hunk 2]
```diff
@@ -65,25 +65,21 @@ Email
...
- * All discussion topics should go to `release drivers mailing list <https://mail.mozilla.org/listinfo/release-drivers>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:69]: Similarly with above, url is https://groups.google.com/a/mozilla.org/g/release-drivers

File: procedures/release-duty/index.rst [hunk 3]
```diff
@@ -65,25 +65,21 @@ Email
...
-
-These other mailing lists will likely have useful discussions and information:
+These mailing lists will likely have useful discussions and information:
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:68]: Nit: we should probably rename this to something like: "Please join the following groups to stay on top of the communication regarding the releases." or alike.

File: procedures/release-duty/index.rst [hunk 4]
```diff
@@ -65,25 +65,21 @@ Email
...
  * `RelEng internal mailing list <release@mozilla.com>`__
- * `Thunderbird release drivers mailing list <https://mail.mozilla.org/listinfo/thunderbird-drivers>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:73]: Idem with above, url is https://groups.google.com/a/mozilla.org/g/thunderbird-drivers

File: procedures/release-duty/index.rst [hunk 5]
```diff
@@ -65,25 +65,21 @@ Email
...
- * `release-automation-notifications-thunderbird mailing list <https://mail.mozilla.org/listinfo/release-automation-notifications-thunderbird>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:74]: This was removed on the 9th of July 2020 as far as I can tell so it's great to clean it up.

File: procedures/release-duty/index.rst [hunk 6]
```diff
@@ -65,25 +65,21 @@ Email
...
  * (optional) `release-automation-notifications-dev mailing list <https://groups.google.com/a/mozilla.com/forum/#!forum/release-automation-notifications-dev>`__
 
 Meetings and Calendars
 ~~~~~~~~~~~~~~~~~~~~~~
 
+See the `Release owners <https://wiki.mozilla.org/Release_Management/Release_owners>`__ list to determine who is ReleaseDuty for each release.
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:75]: Nice 👍

File: procedures/release-duty/index.rst [hunk 7]
```diff
@@ -249,7 +245,7 @@ If a release is blocked. The normal flow is to:
 3. file a ticket
 4. determine which team(s) and person(s) should be escalated.
 
-   a. Searching phonebook is useful for org and ownership charts.
+   a. Searching the People Directory is useful for org and ownership charts.
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:248]: 👍 nice catch

File: procedures/release-duty/index.rst [hunk 8]
```diff
@@ -115,7 +111,7 @@ File an Release Engineering bug under the `Applications: Shipit (backend)
 <https://bugzilla.mozilla.org/enter_bug.cgi?format=__default__&cloned_bug_id=1626312&product=Release%20Engineering&component=Applications%3A%20Shipit%20%28backend%29>`__
```
Comment (MihaiTabara) [inline procedures/release-duty/index.rst:111]: Nit: should we maybe fix `s/an/a` at the line above since we're here?

**Comment by MihaiTabara** [comment]
@gbrownmozilla this is looking good! just a few minor nits here and there to improve things, but we should be good. thanks for taking a stab at this 👍

File: procedures/release-duty/index.rst (hunk 1 above)
Comment (gbrownmozilla) [reply procedures/release-duty/index.rst:68]: Thanks! I missed that somehow. Added back with google groups url.

**Review by MihaiTabara — APPROVED** [review]
